### PR TITLE
openshift.io #1476: Adding notifications about workspaces migration to Multi-Tenant che server

### PR DESCRIPTION
**Pull request by ibuziuk**
https://github.com/openshiftio/openshift.io/issues/1476

Tested locally by applying migration scripts to free-int:

out

### CODE REVIEW

**Comment by ibuziuk** [comment]
[test]

**Comment by ibuziuk** [comment]
@joshuawilson @dlabrecq not sure if this PR is causing this unit test failure:


```
FAILED TESTS:
  AppListComponent
    ✖ should create
      PhantomJS 2.1.1 (Linux 0.0.0)
    Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.

```

But looks like UI is broken since https://ci.centos.org/job/devtools-fabric8-ui-npm-publish-build-master/782/

**Comment by joshuawilson** [comment]
I guess we do need that line, I'll revert it.

**Comment by ibuziuk** [comment]
[test]

**Comment by fabric8cd** [comment]
@ibuziuk fabric8/fabric8-ui:SNAPSHOT-PR-2273-2 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2273-fabric8-ui.badger.fabric8.io

**Comment by ibuziuk** [comment]
While testing locally it turns out to be that notifications are not updated correctly:

image

for `{"multiTenant":true,"running":false}` the message is still `Your Workspaces are loading..." and not `Migrating workspaces to the Multi-Tenant Che server...`
@joshuawilson @dlabrecq Will try to investigate tomorrow, but maybe you have some ideas what is wrong with the PR ?

**Comment by ibuziuk** [comment]
PR has been updated and seems to work just fine

**Comment by ibuziuk** [comment]
[test]

**Comment by bartoszmajsak** [comment]
@joshua would it make sense to cover this new feature in an angular component test?

**Comment by joshuawilson** [comment]
@ibuziuk please rebase

**Comment by ibuziuk** [comment]
@joshuawilson PR has been rebased

**Comment by ibuziuk** [comment]
@joshuawilson looks like the build is misbehaving
```
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! fabric8-ui@0.0.0-development build:prod: `BUILD_VERSION=$(./version_number.js) webpack --config config/webpack.prod.js  --profile --bail`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the fabric8-ui@0.0.0-development build:prod script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above
```

**Comment by ibuziuk** [comment]
[test]

**Comment by ibuziuk** [comment]
[test]

**Comment by ibuziuk** [comment]
`./run_unit_tests.sh` works fine for me locally, so there is smth. wrong with the build

**Comment by ibuziuk** [comment]
[test]

**Comment by fabric8cd** [comment]
@ibuziuk fabric8/fabric8-ui:SNAPSHOT-PR-2273-7 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2273-fabric8-ui.badger.fabric8.io

File: src/app/space/create/codebases/codebases-item/codebases-item.component.ts [hunk 1]
```diff
@@ -58,7 +60,11 @@ export class CodebasesItemComponent implements OnDestroy, OnInit {
    */
   getNotificationMessage(): string {
     if (this.cheState) {
-      return this.cheState.running ? this.cheRunningMessage : this.cheStartingMessage;
+      if (this.cheState.multiTenant) {
+        return this.cheState.running ? this.cheFinishedMultiTenantMigrationMessage : this.chePerformingMultiTenantMigrationMessage;
+      } else {
+        return this.cheState.running ? this.cheRunningMessage : this.cheStartingMessage;
+      }
```
Comment (dlabrecq) [inline src/app/space/create/codebases/codebases-item/codebases-item.component.ts]: The code looks ok to me, but what happens when Che is idle? That is, when does the migration kick in?

Comment (ibuziuk) [reply src/app/space/create/codebases/codebases-item/codebases-item.component.ts]: if che is idled it is not a problem for migration - migration pod will start single-tenant che-server, migrate workspaces to multi-tenant one, and than remove single-tenant deployment

**Comment by fabric8cd** [comment]
@ibuziuk fabric8/fabric8-ui:SNAPSHOT-PR-2273-9 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2273-fabric8-ui.badger.fabric8.io

**Pull request merged**